src/libraries/System.Text.Json/src/System/Text/Json/Reader/JsonReaderHelper.cs

```diff
@@ -383,5 +246,603 @@ public static bool TryGetFloatingPointConstant(ReadOnlySpan<byte> span, out doub
             value = 0;
             return false;
         }
+
+#if PLATFORM_INTRINSICS
+        // Borrowed and modified from SpanHelpers.Byte:
+        // https://github.com/dotnet/runtime/blob/a196d534495a1654ffd158211ca8761b1eba8c05/src/libraries/System.Private.CoreLib/src/System/SpanHelpers.Byte.cs#L916-L1207
+        [MethodImpl(MethodImplOptions.AggressiveOptimization)]
+        private static int IndexOfOrLessThan(ref byte searchSpace, byte value0, byte value1, byte lessThan, int length)
```

Comment:
Any thoughts on ways we we can share the implementation between this and `SpanHelpers`?

This version is much larger due to the Sse2/AdvSimd paths, so I haven't dug into what the modifications were.

Reply:
> This version is much larger due to the Sse2/AdvSimd paths, ...

Is the same size as its copied from the `SpanHelpers` methods which have `AdvSimd` also

https://github.com/dotnet/runtime/blob/4c79d45edd899133cf6d36dc9b4de122ceca72cf/src/libraries/System.Private.CoreLib/src/System/SpanHelpers.Byte.cs#L821-L833

> Any thoughts on ways we we can share the implementation between this and `SpanHelpers`?

As `IndexNotOfAny` was rejected https://github.com/dotnet/runtime/issues/28795#issuecomment-717466792 due to combination explosion I doubt `IndexOfEitherOrLessThan(val0, val1, lessThan)` would be accepted?

Could maybe have a generator for the method in someway..

Reply:
Could maybe have a generator for the method in someway.... Annotate with the conditions?
```
[Simd.Search("=", "=", "<")]
private parital static int IndexOfOrLessThan(ref byte searchSpace, byte value0, byte value1, byte lessThan, int length);
```